### DISCUSSION TITLE
Add to calendar stopped working in LeadPages

### DISCUSSION THREAD

**Discussion by EmadAlfarwan** [add2cal/add-to-calendar-button#451] [Q&A] [answered]
Hi Jens,
We've a printed calendar with QR code in where the user can add the events of each month to mobile calendar.
I use LeadPages.com to generate the webpages where the users can add some events to their mobile calendar
It was working very good, but for some reason stopped working!!
I don't know exactly when it stopped, but I'm sure only few days back or even the problem started today. Because one of my customers raised this issue to customer service team.

here is one of the pages : https://firkahplus.lpages.co/1445-01/ 
you can easily find the add-to-calendar-button but not active!

The way I integrated the code in LeadPages is by adding HTML Widget and paste the code, then it will work.
I added some changes to the style, like using specific Arabic font, here is the code that was working:
```
<table style="margin-left:auto;margin-right:auto;">
    <tr>
        <td>

<script src="https://cdn.jsdelivr.net/npm/add-to-calendar-button@2" async defer> </script>

<add-to-calendar-button 
  styleLight="--font: Din"
  name="بداية العام الهجري 1445"
  startDate="2023-07-19"
  options="'Apple','Google','iCal','Outlook.com','Yahoo','Microsoft365','MicrosoftTeams'"
  listStyle="modal"
  buttonStyle="round"
  size="8"
  label="أضف للتقويم"
  description="تقويم أجواء من فكرة بلس"
></add-to-calendar-button>

          </td>
    </tr>
</table>
```

**After I was informed about the issue, I used the script as is again (you can try the first add-to-calendar button in my web page, and it is obvious because its formatting is different)**
```
<add-to-calendar-button 
  name="بداية العام الهجري 1445"
  startDate="2023-07-19"
  timeZone="Asia/Riyadh"
  options="'Apple','Google','iCal','Outlook.com','Yahoo','Microsoft365','MicrosoftTeams'"
  listStyle="modal"
  buttonStyle="round"
  size="8"
  language="ar"
></add-to-calendar-button>
```

I checked the code here https://add-to-calendar-button.com/#demo , it is working fine.

Now I don't know exactly how to fix this issue.
Thanks

**Comment by jekuer** [comment]
Thanks for bringing this up. It's a bug. Patching it asap.

Btw: You should always set the language to "ar" - this makes the style go right-to-left 😉 .

**Reply by jekuer** [reply, marked as answer]
It got fixed with the latest release v2.4.1 - it usually takes some time, until it is all the way through to the CDN.
If you cannot wait, you can temporarily pin the version like this (revert it after 10 days):

`<script src="https://cdn.jsdelivr.net/npm/add-to-calendar-button@2.4.1" async defer> </script>`

**Reply by EmadAlfarwan** [reply]
Hi Jens,
Thank you so much, this fixed the issue 100%
for Arabic, I tried it but, I'm sure you used Google Translate.
The problem with Google Translate is Literal Translation specially with 1 words that has different meaning. when you used the automatic translation, it used the **Adjective of "Close"** instead of the verb "Close", so the meaning in Arabic didnt look professional, so I had to keep the English and manually change the label to correct Arabic Meaning.

By the way, I can cooperate with you in translating the correct words in Arabic.
Just send me the list of "Messages & Labels"

Regards,
Emad

**Reply by jekuer** [reply]
Some native speaking support would be awesome!!
-> here you can find all the strings: https://github.com/add2cal/add-to-calendar-button/blob/main/src/atcb-i18n.js

**Regarding the language:**
As said, it would make the layout look better.
You can also translate the "Close" with the customLabels option. Simply add something like this:
`customLabels='{"close":"your custom close label text"}'`

But if you could provide better translation, I would also update the core accordingly and then there's no need for the customization 😃
